### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by xerxes-on**
added profile page and all the functionality

### CODE REVIEW

File: src/api/client.js [hunk 1]
```diff
@@ -1,24 +1,10 @@
...
-            config.headers.Authorization = `Bearer ${token.value}`
-        }
-        return config
+    headers: {
+        'Content-Type': 'multipart/form-data',
     },
-    (error) => {
-        return Promise.reject(error)
-    }
-)
```
Comment (volhaliferovich) [inline src/api/client.js:22]: Why did you remove interceptors?

File: src/api/editProfile.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import client from '@/api/client.js'
+
+const edit = (data) => {
+    return client
+        .post('/profile_edit?_method=PUT', data)
```
Comment (volhaliferovich) [inline src/api/editProfile.js:5]: You can use client.put instead if your api supports it

File: src/components/Home/Suggestions.vue [hunk 3]
```diff
@@ -2,16 +2,18 @@
 import { VueperSlide, VueperSlides } from 'vueperslides'
 import 'vueperslides/dist/vueperslides.css'
 
-const props = defineProps({
-    suggestedBooks: {
+defineProps({
+    homeStore: {
```
Comment (volhaliferovich) [inline src/components/Home/Suggestions.vue:6]: I've already asked you to pass suggestions, not homeStore here

File: src/components/Login/LoginForm.vue [hunk 4]
```diff
@@ -9,7 +9,12 @@ import Svg from '@/components/common/Image.vue'
 import ErrorPopup from '@/components/common/ErrorPopup.vue'
 
 const router = useRouter()
+const authStore = useAuthStore()
 const showPassword = ref(false)
+const v$ = useVuelidate(validations, data)
+const error = ref(false)
+const message = ref('Something went wrong')
+
 const data = reactive({
```
Comment (volhaliferovich) [inline src/components/Login/LoginForm.vue:18]: data and validations should go before useVuelidate because you use them there

File: src/components/Login/RegisterForm.vue [hunk 5]
```diff
@@ -10,15 +10,18 @@ import ErrorPopup from '@/components/common/ErrorPopup.vue'
 
 const showPassword = ref(false)
 const router = useRouter()
+const v$ = useVuelidate(validations, data)
+const error = ref(false)
+const response = await authAPI.register(data)
+const passwordRef = toRef(data, 'password')
+const authStore = useAuthStore()
 
 const data = reactive({
```
Comment (volhaliferovich) [inline src/components/Login/RegisterForm.vue:19]: Same here

File: src/components/Profile/UserInfo.vue [hunk 6]
```diff
@@ -1,21 +1,61 @@
...
 import { useProfileStore } from '@/stores/profile.js'
-import { formatDate } from 'compatx'
-import Input from '@/components/common/Input.vue'
+import editProfileApi from '@/api/editProfile.js'
+import { formatDate } from '@/utilities.js'
+import ErrorPopup from '@/components/common/ErrorPopup.vue'
 
 const profileStore = useProfileStore()
 const user = profileStore.user.user
 const profile_pic = user.profile_picture !== null ? user.profile_picture : 'src/assets/images/cat.jpg'
```
Comment (volhaliferovich) [inline src/components/Profile/UserInfo.vue:10]: I have asked you to change it to computed

File: src/components/Profile/UserInfo.vue [hunk 7]
```diff
@@ -1,21 +1,61 @@
...
+
+function removeNullKeys(obj) {
+    for (const key in obj) {
+        if (obj[key] === null) {
+            delete obj[key]
+        }
+    }
+}
+
+const errorMessage = ref('')
```
Comment (volhaliferovich) [inline src/components/Profile/UserInfo.vue:34]: Move it to the reactive vars

**Review by volhaliferovich — CHANGES_REQUESTED** [review]

File: src/api/client.js (hunk 1 above)
Comment (xerxes-on) [reply src/api/client.js:22]: I guess it happened while resolving the conflicts

File: src/api/editProfile.js (hunk 2 above)
Comment (xerxes-on) [reply src/api/editProfile.js:5]: it didnt work turns out in laravel there is a problem with that

**Review by volhaliferovich — APPROVED** [review]
